### PR TITLE
Bump Go version to 1.23

### PR DESCRIPTION
**Pull request by maranqz**
1. bump up go version
2. fix test running.
Currently tests always pass because no exit code is sent for tests.
I fixed this, and also the driver tests only run on go versions and higher that are installed in go.mod.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [2.0.0-rc-9] - 2024-08-09
+
+### Changes
+
+- Bumped tests up to go1.22
```
Comment (maranqz) [inline CHANGELOG.md:12]: Goleak message should be fixed. If there is message, test should fail
https://github.com/avito-tech/go-transaction-manager/actions/runs/10226853987/job/28297721109?pr=107#step:8:85

Comment (maranqz) [reply CHANGELOG.md:12]: Resolved by `defer mongodbClient.Disconnect(ctx)`

File: drivers/mongo/example_test.go [hunk 2]
```diff
@@ -24,8 +24,9 @@ func Example() {
 	ctx := context.Background()
 
 	client, err := mongo.Connect(ctx, options.Client().
-		ApplyURI("mongodb://127.0.0.1:27017"))
+		ApplyURI("mongodb://127.0.0.1:27017/?directConnection=true"))
 	checkErr(err)
+	defer client.Disconnect(ctx)
```
Comment (maranqz) [inline drivers/mongo/example_test.go:29]: // TODO

If goleaks show problems, github actions should fail. Now actions is pass even if goleaks alert about a problem.

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -53,7 +52,7 @@ jobs:
         run: make test
 
   tests-integration:
-    needs: [tests-units]
+    # needs: [tests-units]
```
Comment (maranqz) [inline .github/workflows/main.yaml:55]: Rollback

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,7 +5,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
-## [2.0.0-rc-9] - 2024-08-03
+## [2.0.0-rc-9.5] - 2024-08-09
```
Comment (maranqz) [inline CHANGELOG.md:8]: Add explanation of fix

File: README.md [hunk 5]
```diff
@@ -167,3 +167,16 @@ func (r *repo) Save(ctx context.Context, u *user) error {
...
+- [golangci-lint](https://golangci-lint.run/welcome/install/)
+
+### Local Running
+
+To run database by docker, there is [docker-compose.yaml](trm/drivers/test/docker-compose.yaml).
+```bash
+docker compose -f trm/drivers/test/docker-compose.yaml up
+```
+
+For full GitHub Actions run, you can use [act](https://github.com/nektos/act).
```
Comment (maranqz) [inline README.md:182]: Add information how to bump up the version.

File: README.md [hunk 6]
```diff
@@ -167,3 +167,16 @@ func (r *repo) Save(ctx context.Context, u *user) error {
...
+
+### Requirements
+
+- [golangci-lint](https://golangci-lint.run/welcome/install/)
+
+### Local Running
+
+To run database by docker, there is [docker-compose.yaml](trm/drivers/test/docker-compose.yaml).
+```bash
+docker compose -f trm/drivers/test/docker-compose.yaml up
```
Comment (maranqz) [inline README.md:179]: Prove that old version vulnerabilities will not happen in user code.

**Comment by AlexeyKupershtokh** [comment]
Hey Ilia.

Please consider that gh action started upgrading go 1.21 to 1.22 automatically.
https://github.com/avito-tech/go-transaction-manager/actions/runs/10230981994/job/28306419497#step:3:14

There's an article and a possible fix for this: https://brandur.org/fragments/go-version-matrix

File: README.md (hunk 6 above)
Comment (maranqz) [reply README.md:187]: added here 

[trm/internal/prove_receiving_updates_without_vulnerabilities/README.md](https://github.com/avito-tech/go-transaction-manager/pull/107/files#diff-57dea7d98e4da9e094f098dff3a4bbef23ab16550862c26b9c7e245fa914ad1a)

File: README.md [hunk 7]
```diff
@@ -162,8 +170,44 @@ func (r *repo) Save(ctx context.Context, u *user) error {
...
 * To run all tests use `make go.test` or `make go.test.with_real_db` for integration tests.
+
+To run database by docker, there is [docker-compose.yaml](trm/drivers/test/docker-compose.yaml).
+```bash
+docker compose -f trm/drivers/test/docker-compose.yaml up
+```
+
+For full GitHub Actions run, you can use [act](https://github.com/nektos/act).
+
+#### Running old go versions
```
Comment (maranqz) [inline README.md:192]: Explain how to solve version conflict

File: README.md [hunk 8]
```diff
@@ -42,6 +42,14 @@ For example `go get github.com/avito-tech/go-transaction-manager/drivers/sqlx/v2
 The library is compatible with the most recent two versions of Go.
```
Comment (maranqz) [inline README.md:42]: Fix mongo tests

https://github.com/avito-tech/go-transaction-manager/actions/runs/10636006087/job/29487001688?pr=107#step:8:87

**Review by vladjong — APPROVED** [review]

**Review by Khranovskiy — APPROVED** [review]

**Pull request merged**